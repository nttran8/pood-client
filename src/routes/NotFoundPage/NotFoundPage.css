.NotFoundPage {
  padding-top: 120px;
}

.FourOhFour {
  font-size: 6rem;
  color: var(--color-triad-r);
}

.NotFoundPage > p {
  font-weight: 600;
  margin-bottom: 30px;
}

.FourOhFourButton {
  width: 150px;
  margin: auto;
}

@media only screen and (min-width: 576px) {
  .FourOhFour {
    font-size: 10rem;
  }

  .NotFoundPage > p {
    font-size: 18px;
  }

  .FourOhFourButton {
    width: 180px;
    margin: auto;
  }

}