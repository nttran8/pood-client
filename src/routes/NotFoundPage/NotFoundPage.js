// Libraries
import React, { Component } from 'react';
import { CreateButton } from '../../components/Utils/Utils';

// Style
import './NotFoundPage.css';

export default class NotFoundPage extends Component {
  static defaultProps = {
    history: {
      push: () => {}
    }
  };

  handleHomePageRedirect = () => {
    // 
    const { history } = this.props;
    history.push('/');
  };

  render() {
    return (
      <section className='NotFoundPage'>
        <h2 className='FourOhFour'>404</h2>
        <p>Poop. There's nothing here...</p>
        <div className='FourOhFourButton'>
          <CreateButton onClick={this.handleHomePageRedirect}>Back to home</CreateButton>
        </div>
      </section>
    );
  }
}
