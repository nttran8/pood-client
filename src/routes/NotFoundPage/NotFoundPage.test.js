import React from "react";
import TestRenderer from "react-test-renderer";
import NotFoundPage from "./NotFoundPage";

it(`Renders NotFoundPage UI as expected`, () => {
  const tree = TestRenderer.create(<NotFoundPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
