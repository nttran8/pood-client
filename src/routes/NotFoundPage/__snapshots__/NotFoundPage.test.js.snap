// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders NotFoundPage UI as expected 1`] = `
<section
  className="NotFoundPage"
>
  <h2
    className="FourOhFour"
  >
    404
  </h2>
  <p>
    Poop. There's nothing here...
  </p>
  <div
    className="FourOhFourButton"
  >
    <button
      className="button "
      onClick={[Function]}
    >
      Back to home
    </button>
  </div>
</section>
`;
